src/gcovr/formats/gcov/parser/text.py

```diff
@@ -305,6 +305,19 @@ def parse_coverage(
         except Exception as ex:  # pylint: disable=broad-except
             lines_with_errors.append((raw_line, ex))
 
+    if not any(isinstance(line, _FunctionLine) for line, _ in tokenized_lines):
+        file_for_message = (
+            "\n  ".join(" -> ".join(file_tuple) for file_tuple in sorted(data_filename))
+            if isinstance(data_filename, set)
+            else data_filename
+        )
+        raise RuntimeError(
+            f"No function line found in gcov file:\n   {file_for_message}\n"
```

Comment:
Note that this has three spaces after the newline before printing `file_for_merge`, while `file_for_merge` uses two spaces after the newline when there are multiple files.